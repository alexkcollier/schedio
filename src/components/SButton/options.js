export {
  colors,
  allShadeOptions,
  colorNames,
} from '@/utils/component-options'

export const sizes = [
  'small',
  'regular',
  'large',
]

export const types = [
  'outlined',
  'text',
]
