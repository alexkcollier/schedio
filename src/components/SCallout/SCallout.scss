.callout {
  background: color('ice');
  border-left: border();
  border-left-width: border-width('thickest');
  border-radius: border-radius('small');
  padding: spacing();

  & + & {
    margin-top: spacing();
  }

  &--success {
    background: lighten(color('green', 'lighter'), 45%);
    border-left-color: color('green');
  }

  &--warning {
    background: lighten(color('gold', 'lighter'), 13%);
    border-left-color: color('gold');
  }

  &--error {
    background: lighten(color('red', 'lighter'), 10%);
    border-left-color: color('red');
  }
}
