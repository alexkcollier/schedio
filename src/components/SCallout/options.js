export const calloutTypes = ['', 'success', 'warning', 'error']
