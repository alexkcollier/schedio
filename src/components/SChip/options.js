import { props } from '@/assets/styles/tokens/dist/tokens.raw.json'

export const colors = [
  ...Object.values(props).reduce((colorList, { category, name }) => {
    if (category === 'background-color') colorList.add(name.split('-')[0])

    return colorList
  }, new Set()),
].sort()
