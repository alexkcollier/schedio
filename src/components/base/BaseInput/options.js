export const allowed = ['email', 'text', 'password', 'url', 'search', ''].sort()
