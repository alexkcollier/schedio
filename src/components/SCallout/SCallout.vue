<template>
  <div
    class="callout"
    :class="classList"
  >
    <slot />
  </div>
</template>

<script>
import { calloutTypes } from './options'

export default {
  name: 'SCallout',

  props: {
    calloutType: {
      type: String,
      default: calloutTypes[0],
      validator: v => {
        return (
          calloutTypes.includes(v) ||
          console.error(`\`calloutType\` should be one of ${calloutTypes.join(', ')}`)
        )
      }
    }
  },

  computed: {
    classList() {
      return {
        ...(this.calloutType && { [`callout--${this.calloutType}`]: !!this.calloutType })
      }
    }
  }
}
</script>
