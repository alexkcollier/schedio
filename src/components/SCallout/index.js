import SCallout from './SCallout.vue';

export { SCallout };

export default {
  install: Vue => Vue.component(SCallout.name, SCallout),
};
