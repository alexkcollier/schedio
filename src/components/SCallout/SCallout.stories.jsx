import { SCallout } from '@/components/SCallout'
import { SHeading } from '@/components/SHeading'
import PropList from '@@/docs/components/PropList'
import { generateHeading } from '@/utils/stories/render-functions'
import { select, text } from '@storybook/addon-knobs'
import jeffsum from 'jeffsum'
import { types } from './options'

const defaultSlotPlaceholder = jeffsum(5, 'sentences')

export default {
  title: 'Components|SCallout',

  parameters: {
    jest: 'SCallout',
  },
}

export const callout = () => ({
  props: {
    props: {
      default: {
        type: select('type', types, types[0], 'Optional Props'),
      },
    },
    slots: {
      default: {
        default: text('Default slot', defaultSlotPlaceholder, 'Slots'),
      },
    },
  },

  render (h) {
    const { slots, props } = this.$props
    return (
      <div>
        <SHeading>Callout</SHeading>
        <p>Callouts are used to place special emphasis on text and call attention to it.</p>

        <SHeading level="2">Example</SHeading>
        <SCallout {...{ props }}>{slots.default}</SCallout>

        {SCallout.props && <PropList component={SCallout} />}
      </div>
    )
  },
})

callout.story = {
  name: 'Callout',
}

export const calloutTypes = () => ({
  render (h) {
    return h('div', [
      generateHeading(h, { content: 'Callout Types' }),
      types.map(type => [
        generateHeading(h, { level: 3, content: type || 'Base' }),
        h(SCallout, { props: { type } }, jeffsum(5)),
      ]),
    ])
  },
})

calloutTypes.story = {
  name: 'Callout Types',

  parameters: {
    options: {
      showPanel: false,
    },
  },
}
