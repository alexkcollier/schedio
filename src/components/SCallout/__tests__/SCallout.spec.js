import { shallowMount } from '@vue/test-utils';
import SCallout from '@/components/SCallout/SCallout.vue';
import { types } from '../options';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);

describe('SCallout.vue', () => {
  const errorSpy = jest.spyOn(global.console, 'error').mockImplementation(() => { });
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(SCallout);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('renders correctly', async () => {
    const wrapper = shallowMount(SCallout);

    expect(wrapper.html()).toMatchSnapshot();
    expect(await axe(wrapper.html())).toHaveNoViolations();
  });

  Promise.all(
    // eslint-disable-next-line array-callback-return
    types.map((type) => {
      it(`can have type ${type}`, async () => {
        wrapper.setProps({ type });

        expect(await axe(wrapper.html())).toHaveNoViolations();
        expect(errorSpy).not.toHaveBeenCalled();
        expect(wrapper.html()).toMatchSnapshot();
      });
    }),
  );

  it('validates the `type`', () => {
    shallowMount(SCallout, {
      propsData: {
        type: 'not a type',
      },
    });

    expect(errorSpy).toHaveBeenCalled();
  });

  it('accepts slot content', () => {
    const slotText = 'I am slot text';
    const slotWrapper = shallowMount(SCallout, {
      slots: {
        default: slotText,
      },
    });

    expect(slotWrapper.text()).toBe(slotText);
    expect(slotWrapper.html()).toMatchSnapshot();
  });
});
