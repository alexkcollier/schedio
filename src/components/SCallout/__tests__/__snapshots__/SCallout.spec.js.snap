// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCallout.vue accepts slot content 1`] = `<div class="callout">I am slot text</div>`;

exports[`SCallout.vue can have type  1`] = `<div class="callout"></div>`;

exports[`SCallout.vue can have type error 1`] = `<div class="callout callout--error"></div>`;

exports[`SCallout.vue can have type success 1`] = `<div class="callout callout--success"></div>`;

exports[`SCallout.vue can have type warning 1`] = `<div class="callout callout--warning"></div>`;

exports[`SCallout.vue renders correctly 1`] = `<div class="callout"></div>`;
